storage/innobase/handler/ha_innodb.cc

```diff
@@ -22653,3 +22689,21 @@ ib_push_frm_error(
 		break;
 	}
 }
+
+/** Calculate aligned buffer pool size based on srv_buf_pool_chunk_unit,
+if needed.
+@param[in]	size	size in bytes
+@return	aligned size */
+ulint
+buf_pool_size_align(
+	ulint	size)
+{
+  const ib_uint64_t	m = ((ib_uint64_t)srv_buf_pool_instances) * srv_buf_pool_chunk_unit;
+  size = ut_max((size_t) size, (size_t) MYSQL_SYSVAR_NAME(buffer_pool_size).min_val);
+
+  if (size % m == 0) {
+    return(size);
+  } else {
+    return (ulint)((size / m + 1) * m);
+  }
+}
```

Comment:
We could ask why `innodb_buffer_pool_size` has been declared as `longlong` instead of `size_t` in the first place. But, I think that it was fixed in a later version already.

I would avoid old InnoDB style excessive line breaks and `{}` around single-statement blocks in new code.
```c++
ulint buf_pool_size_align(ulint size)
{
  const size_t m= …;
  size= std::max(size, size_t(MYSQL_SYSVAR_NAME(buffer_pool_size).min_val);
  return size % m
    ? size
    : (size / m + 1) * m;
}
```
Note: `UT_CALC_ALIGN()` can’t be used, because `innodb_buffer_pool_chunk_size` is not necessarily a power of 2.